diff --git a/node_modules/@rosen-bridge/discord-notification/dist/DiscordNotification.js b/node_modules/@rosen-bridge/discord-notification/dist/DiscordNotification.js
index 85c6321..ba00a1f 100644
--- a/node_modules/@rosen-bridge/discord-notification/dist/DiscordNotification.js
+++ b/node_modules/@rosen-bridge/discord-notification/dist/DiscordNotification.js
@@ -13,7 +13,7 @@ class DiscordNotification extends AbstractNotification {
      * @param message
      */
     sendMessage = (message) => {
-        this.client.send({
+        return this.client.send({
             content: message.replace(/^ +| +$/gm, ''), // trim all lines
         });
     };
@@ -23,7 +23,8 @@ class DiscordNotification extends AbstractNotification {
      * @param message
      */
     error = async (title, message) => {
-        this.sendMessage(`
+        return this.sendMessage(`
       ## ❌ ${title}
       ${message}
       `);
@@ -34,7 +35,7 @@ class DiscordNotification extends AbstractNotification {
      * @param message
      */
     warning = async (title, message) => {
-        this.sendMessage(`
+        return this.sendMessage(`
       ## ⚠️ ${title}
       ${message}
       `);
@@ -45,7 +46,7 @@ class DiscordNotification extends AbstractNotification {
      * @param message
      */
     info = async (title, message) => {
-        this.sendMessage(`
+        return this.sendMessage(`
       ## ℹ️ ${title}
       ${message}
       `);
@@ -56,7 +57,7 @@ class DiscordNotification extends AbstractNotification {
      * @param message
      */
     success = async (title, message) => {
-        this.sendMessage(`
+        return this.sendMessage(`
       ## ✅ ${title}
       ${message}
       `);
@@ -67,7 +68,7 @@ class DiscordNotification extends AbstractNotification {
      * @param message
      */
     raw = async (title, message) => {
-        this.sendMessage(`
+        return this.sendMessage(`
       ## ${title}
       ${message}
       `);
