diff --git a/node_modules/winston/lib/winston/logger.js b/node_modules/winston/lib/winston/logger.js
index 243cb1e..1c1344b 100644
--- a/node_modules/winston/lib/winston/logger.js
+++ b/node_modules/winston/lib/winston/logger.js
@@ -246,7 +246,7 @@ class Logger extends Transform {
           message: msg
         });
 
-        if (meta.message) info.message = `${info.message} ${meta.message}`;
+        if (meta.message) info.message = `${info.message} ${JSON.stringify(meta.message)}`;
         if (meta.stack) info.stack = meta.stack;
 
         this.write(info);
